### ISSUE TITLE
Allow to search for parts of words

### ISSUE THREAD

**Issue by yennor** [vector-im/element-web#20207]
#### What would you like to do?
If I enter a search string, I want every chat message to be shown where that string appears even if it's only part of a word. Right now it only searchs for whole words.

#### Why would you like to do it?
In english it halfway works looking only for whole words (even there it's problematic, plural, past ...), but in languages like german where wordendings (also nouns) can change a lot and in different ways depending on the context it's just impossible to search for every possibility. There you usually just want to search for the word base (hoping that this one didn't change, which could also happen but not that frequently).

#### How would you like to achieve it?
well. also show results where the search string is only part of a word. you could also make a check box for "only find whole words)

**Comment by t3chguy** [comment]
This behaviour is dependent on your Matrix server, on matrix.org it uses Postgres FTS

**Comment by t3chguy** [comment]
The server is only used for search of cleartext events, local search of encrypted rooms is only possible in Element Desktop which uses seshat https://github.com/matrix-org/seshat

**Comment by yennor** [comment]
ok, in that case my feature request is for element-desktop. I completely forgot that I never search for anything on my phone, since it just doesn't work there... But like that with only supporting whole word search in general I also can't find anything on element-desktop...
so you're telling me I should make a feature request on seshat? or does it already support it and element just makes a call looking only for whole words?

**Comment by t3chguy** [comment]
Most full text search indexers only support full word search because that's how they build the index, partial-word search wouldn't be as efficiently indexable

**Comment by yennor** [comment]
I understand that. It doesn't change my feature request though. In certain languages it's very difficult to search only on whole words (In swiss german it's worse, wordendings can change depending on from which region the writer is).

**Comment by t3chguy** [comment]
Right, but FTS also has a feature called language stemming, where it doesn't index the exact word but rather its stem. So `worrying` would be indexed as `worry`, same with `worried`, `worriedly` etc, so all of those map to any of them.

So maybe what you want is better language stemming support in Seshat and/or Synapse rather than "Allow to search for parts of words"

**Comment by yennor** [comment]
I don't know how well that works with concatenating words? In german you can concatenate any amount of nouns. "House Roof Window" would be hausdachfenster. Because of that I regularly get problems with spellchecking programs, even in 2021.
And I'm pretty sure there are a looot of spoken languages, where nobody ever created a proper grammar support (getting all the word stemming and everything). Swiss german is one of them. While officially it is not a written language, we write personal e-mails, message, chats etc. in it. But no orthography/grammar support exists. Even in germany there are a lot of dialects which are officially only spoken, but in some regions people write messages in them. Stemming support would be difficult.
In the region I am right now, there are a lot of indigenous people, which also write all their messages in their own languages. I'm pretty sure no stemming support exists for those either.  So I guess stemming support is nice, actually great for big official languages (Even makes searches arguably a lot easier there).
But for all the small unsupported languages and dialects it won't work. And in a lot of places in the world where I've been, people write messages in their own language/dialect as long as it is no official communication.
